subcommands/waves/init.go

```diff
@@ -0,0 +1,156 @@
+package waves
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	tuf "github.com/theupdateframework/notary/tuf/data"
+)
+
+func init() {
+	initCmd := &cobra.Command{
+		Use:   "init <wave> <version> [<tag>]",
+		Short: "Create a new wave for a given targets version.",
+		Long: `Create a new wave for a given targets version.",
+Optionally, provide the tag to use for these targets ('master' by default).
+In any case, original tags of these targets are ignored.
+
+This command only initializes a wave, but does not provision its updates to devices.
+Use a "fioctl wave rollout <wave> <group>" to trigger updates of this wave to a device group.
+Use a "fioctl wave complete <wave>" to update all devices (make it globally available).
+Use a "fioctl wave cancel <wave> to cancel a wave (make it no longer available).`,
+		Run:  doInitWave,
+		Args: cobra.RangeArgs(2, 3),
+	}
+	cmd.AddCommand(initCmd)
+	initCmd.Flags().IntP("expires-days", "e", 0, `Role expiration in days; default 365.
+The same expiration will be used for production targets when a wave is complete.`)
+	initCmd.Flags().StringP("expires-at", "E", "", `Role expiration date and time in RFC 3339 format.
+The same expiration will be used for production targets when a wave is complete.
+When set this value overrides an 'expires-days' argument.
+Example: 2020-01-01T00:00:00Z`)
+	initCmd.Flags().BoolP("dry-run", "d", false, "Don't create a wave, print it to standard output.")
+}
+
+func doInitWave(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	name, version, tag := args[0], args[1], "master"
+	if len(args) > 2 {
+		tag = args[2]
+	}
+	intVersion, err := strconv.ParseInt(version, 10, 32)
+	subcommands.DieNotNil(err, "Version must be an integer")
+	expires := readExpiration(cmd)
+	dryRun, _ := cmd.Flags().GetBool("dry-run")
+	logrus.Debugf("Creating a wave %s for factory %s targets version %s and new tag %s expires %s",
+		name, factory, version, tag, expires.Format(time.RFC3339))
+
+	new_targets, err := api.TargetsList(factory, version)
+	subcommands.DieNotNil(err)
+	if len(new_targets) == 0 {
+		subcommands.DieNotNil(fmt.Errorf("No new targets of a given version"))
```

Comment:
a little nitpick, but i just hit this by accident and found this message confusing. I think something like "Target version %d does not exist".